Client connection states

Allen-Glass opened the pull request:
## PR Checklist
<!-- Insert "x" inside square brackets to check item. -->

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:


## What is the current behaviour?
We have no way to track a client's connection state to the coop server


## What is the new behaviour?
A way to track each client's connection state within the coop server.
Resolves #289 and #447
Added a simple event listener to track where a client is at when connected to the coop server. Building block for more complicated server connection state behavior

![image](https://user-images.githubusercontent.com/15619189/212504614-c8ee4507-150b-4e50-aba1-42d849a4b199.png)


## Other information

garrettluskey commented on source/Coop.Core/Server/Connections/Connection.cs:
```diff
@@ -15,7 +15,7 @@ public class Connection : IConnection
         {
             Peer = peer;
-            State = new InitialConnectionState(this);
+            //State = new InitialConnectionState(this);
```
Why is this commented out?

garrettluskey replied:
Isn't this supposed to be where the state initializes?

garrettluskey commented on source/Coop.Core/Server/Connections/ConnectionManager.cs:
```diff
@@ -4,6 +4,7 @@
 namespace Coop.Core.Server.Connections
 {
+
     public readonly struct ClientConnectedMessage
```
To conform with naming convention, remove "Message" as a postfix

garrettluskey replied:
Same with ClientDisconnectedMessage

garrettluskey commented:
Sorry to change stuff up on you, but 2 or 3 states were added to the [State Diagram](https://drive.google.com/file/d/1KhiE62qUYYlH66fsh3BbWlmzo20UmGGS/view?usp=sharing). Please add these :)

garrettluskey commented on source/Coop.Core/Server/Connections/IConnectionState.cs:
```diff
@@ -1,13 +1,27 @@
 
 namespace Coop.Core.Server.Connections
 {
     public interface IConnectionState : IState
```
This will be required to change with the new connection state changes

garrettluskey commented on source/Coop.Core/Server/Connections/ServerStateModule.cs:
```diff
@@ -0,0 +1,16 @@
+
+namespace Coop.Core.Server.Connections
+{
+    internal class ServerStatesModule : Module
```
What is the justification for this module? Was it meant to be removed? PlayerConnectionStates does not exist from what I can tell.

garrettluskey commented on source/Coop.Core/Server/Connections/States/ConnectionLogic.cs:
```diff
@@ -0,0 +1,44 @@
+    public class ConnectionLogic : IConnectionLogic
+    {
+        public ILogger Logger { get; }
+        public ICoopClient NetworkClient { get; }
```
ICoopClient is only client side. Connection states are only server side. This could change to peer but it is never used so I recommend removing.

garrettluskey commented on source/Coop.Core/Server/Connections/States/InitialConnectionState.cs:
```diff
@@ -1,14 +1,29 @@
 namespace Coop.Core.Server.Connections.States
 {
-    internal class InitialConnectionState : IConnectionState
+    public class InitialConnectionState : ConnectionStateBase, IConnectionState
```
IConnectionState is not needed as ConnectionStateBase already inherits that interface

garrettluskey commented on source/Coop.Tests/Server/Connections/States/ClientStateOrchestratorTests.cs:
```diff
@@ -0,0 +1,96 @@
+        [Fact]
+        public void PlayerDisconnected_RemovePlayer()
+        {
+            messageBroker.Publish(this, new ResolveCharacter(_playerId));
```
ResolveCharacter is for getting the controlled party/hero Guid from the server if it already exists (assosiated with this connection). ClientConnected should add a new player.

garrettluskey replied:
Also new states will likely change most of these tests

garrettluskey commented on source/Coop.Core/Server/Connections/States/ResolveCharacterState.cs:
```diff
@@ -0,0 +1,29 @@
+{
+    public class ResolveCharacterState : ConnectionStateBase
+    {
+        public ResolveCharacterState(IConnectionLogic connectionLogic, IMessageBroker messageBroker) 
```
This class is not completable until #315 is complete. Once the new states are implemented it can be approved.

garrettluskey commented on source/MissionTestMod/MissionTestMod.csproj.user:
```diff
@@ -0,0 +1,9 @@
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
+    <StartArguments>/singleplayer /client _MODULES_%2aNative%2aSandBoxCore%2aSandBox%2aStoryMode%2aMissionTestMod%2a_MODULES_</StartArguments>
+    <StartAction>Program</StartAction>
+    <StartProgram>C:\Users\allen\source\repos\BannerlordCoop\mb2\bin\Win64_Shipping_Client\Bannerlord.exe</StartProgram>
```
Convert this to relative paths or remove file from merge

Allen-Glass replied:
good eye! will update

garrettluskey commented on source/Coop.Tests/Server/Connections/States/ResolveCharacterStateTests.cs:
```diff
@@ -0,0 +1,40 @@
+
+namespace Coop.Tests.Server.Connections.States
+{
+    public class ResolveCharacterStateTests : CoopTest
```
This is missing a test to transfer to CreateCharacterState

EgardA committed:
Merge branch 'BladeData-serialization' into taverns-serialization

EgardA committed:
BannerEffectSerialization

EgardA committed:
BasicCultureObjectSerializer

EgardA committed:
ItemModifierSerialization

zzzzzzzzott committed:
test

garrettluskey committed:
Merge pull request #358 from Bannerlord-Coop-Team/Mat3Serialization

Mat3Serialization

garrettluskey committed:
Merge pull request #361 from Bannerlord-Coop-Team/BasicCultureObjectSerialization

BasicCultureObjectSerializer

garrettluskey committed:
Merge branch 'taverns-serialization' into BannerEffectSerialization

garrettluskey committed:
Merge pull request #360 from Bannerlord-Coop-Team/BannerEffectSerialization

BannerEffectSerialization

garrettluskey committed:
Merge branch 'taverns-serialization' into taverns-serialization

garrettluskey committed:
Merge branch 'taverns-serialization' into CraftingTemplateSerialization

garrettluskey committed:
Merge pull request #357 from Bannerlord-Coop-Team/CraftingTemplateSerialization

CraftingTemplateSerialization

EgardA committed:
excludes

zzzzzzzzott committed:
CachedData update

zzzzzzzzott committed:
Merge branch 'taverns-serialization' of https://github.com/zzzzzzzzott/BannerlordCoop into taverns-serialization

zzzzzzzzott committed:
Merge pull request #1 from zzzzzzzzott/taverns-serialization-ext

test

garrettluskey committed:
Merge branch 'taverns-serialization' into taverns-serialization

zzzzzzzzott committed:
Update source/GameInterface.Tests/Serialization/SerializerTests/SaddleComponentSerializationTest.cs

Co-authored-by: Garrett Luskey <garrett.luskey@gmail.com>

garrettluskey committed:
Merge branch 'taverns-serialization' into ItemModifierSerialization

garrettluskey committed:
Merge pull request #362 from Bannerlord-Coop-Team/ItemModifierSerialization

ItemModifierSerialization

garrettluskey committed:
Added hashset support

zzzzzzzzott committed:
Resvolves #337, $335, #328, #341, #340, #336.

Also fixed some test names.

garrettluskey committed:
Merge branch 'taverns-serialization' into 346-create-monsterbinarypackage

garrettluskey committed:
Changed monster to use string id

zzzzzzzzott committed:
Removed changes to ItemModifier

EgardA committed:
PolicyObject, SkillObject, CharacterSkills

EgardA committed:
Updated to 2/3 non-default values

EgardA committed:
Some serializers

garrettluskey committed:
Update MonsterBinaryPackage.cs

garrettluskey committed:
Merge pull request #403 from Bannerlord-Coop-Team/346-create-monsterbinarypackage

MonsterBinaryPackage

zzzzzzzzott committed:
More thorough tests. Also added WeaponComponent binary package.

SaddleComponent and WeaponComponent tests are not passing.

zzzzzzzzott committed:
Patches some bugs.

SaddleComponent still being hard.

garrettluskey committed:
Merge branch 'taverns-serialization' into pr/359

garrettluskey committed:
Fixed backend issues

WakooMan committed:
Added ItemRostre,ItemRosterElement,Hideout,EquipmentElement,Equipment and BanditParty BinaryPackage.

WakooMan committed:
Added Test files for everything, Hideout Serialization and BanditPartyComponent Serialization works fine.

garrettluskey committed:
Update SaddleComponentBinaryPackage.cs

garrettluskey committed:
Merge pull request #359 from zzzzzzzzott/taverns-serialization

TradeItemComponent, TextObject, SaddleComponent, CraftingPiece, ItemObject, MatrixFrame, WeaponComponentData, and WeaponDesignElement serialization

WakooMan committed:
Merged csprojs and current state of taverns-serialization

WakooMan committed:
Tests Written, EquipmentSerialization and ItemRosterSerialization doesn't work.

WakooMan committed:
Changed BinaryPackageFactory methods a llittle bit.

WakooMan committed:
EquipmentSerializationTest is fine now.

garrettluskey committed:
Update ItemRosterSerializationTest.cs

garrettluskey committed:
Update BinaryPackageFactory.cs

garrettluskey committed:
Merge branch 'taverns-serialization' into Hasted-Serialization

garrettluskey committed:
Update CharacterSkillsSerializationTest.cs

garrettluskey committed:
Updated test

garrettluskey committed:
Merge pull request #404 from Bannerlord-Coop-Team/Hasted-Serialization

PolicyObject, SkillObject, CharacterSkills serializers

garrettluskey committed:
Added array support for Serializable check

garrettluskey committed:
Merge branch 'taverns-serialization' into pr/406

EgardA committed:
Merge branch 'taverns-serialization' into MoreSerializers

EgardA committed:
updated

WakooMan committed:
Done requested changes

EgardA committed:
Update source/GameInterface/Serialization/Impl/CharacterTraitsBinaryPackage.cs

Co-authored-by: Garrett Luskey <garrett.luskey@gmail.com>

WakooMan committed:
HideoutSerializationTest updated, requested changes added.

garrettluskey committed:
Merge branch 'taverns-serialization' into WakooSerializers

garrettluskey committed:
Merge pull request #406 from WakooMan/WakooSerializers

Equipment,EquipmentElement,ItemRoster,ItemRosterElement,Hideout and BanditPartyComponent Binary Packages Created and Tested

garrettluskey committed:
Merge branch 'taverns-serialization' into MoreSerializers

garrettluskey committed:
Merge pull request #405 from Bannerlord-Coop-Team/MoreSerializers

Village, Town, Settlement, CharacterTraits, CultureObject, PerkObject, TraitObject & BodyProperties

Allen-Glass committed:
refactoring message broker

EgardA committed:
update

garrettluskey committed:
initial commit

garrettluskey committed:
Update GameInterface.Tests.csproj

garrettluskey committed:
Create DictTest.cs

garrettluskey committed:
the big clean

garrettluskey committed:
Merge branch 'taverns-serialization' into ItemModifierGroupBinaryPackage

EgardA committed:
most things fixed

garrettluskey committed:
Update WeaponDesignBinaryPackage.cs

garrettluskey committed:
Update WeaponDesignBinaryPackage.cs

garrettluskey committed:
Merge branch 'taverns-serialization' into H-Serialization

garrettluskey committed:
removed unused imports

garrettluskey committed:
Merge pull request #407 from Bannerlord-Coop-Team/H-Serialization

WeaponDesign serializer

garrettluskey committed:
Merge branch 'taverns-serialization' into ItemModifierGroupBinaryPackage

garrettluskey committed:
Merge pull request #409 from Bannerlord-Coop-Team/ItemModifierGroupBinaryPackage

Item modifier group binary package

garrettluskey committed:
backend updates with troop roster binary package

garrettluskey committed:
Merge branch 'taverns-serialization' into TroopRosterBinaryPackage

EgardA committed:
update

garrettluskey committed:
Merge pull request #411 from Bannerlord-Coop-Team/TroopRosterBinaryPackage

TroopRosterBinaryPackage

garrettluskey committed:
Merge pull request #410 from Bannerlord-Coop-Team/H-Serialization

MaterialPropertySerialization

garrettluskey committed:
Initial commit

garrettluskey committed:
Merge branch 'taverns-serialization' into HorseComponentBinaryPackage

garrettluskey committed:
Initial commit

garrettluskey committed:
Update ReflectionExtensions.cs

garrettluskey committed:
Merge branch 'taverns-serialization' into ArmorComponentBinaryPackage

garrettluskey committed:
Initial Commit

garrettluskey committed:
Merge pull request #412 from Bannerlord-Coop-Team/HorseComponentBinaryPackage

HorseComponentBinaryPackage

garrettluskey committed:
Merge branch 'taverns-serialization' into BannerComponentBinaryPackage

garrettluskey committed:
Merge branch 'taverns-serialization' into ArmorComponentBinaryPackage

garrettluskey committed:
Merge pull request #413 from Bannerlord-Coop-Team/ArmorComponentBinaryPackage

ArmorComponentBinaryPackage

garrettluskey committed:
Update BannerComponentSerializationTest.cs

garrettluskey committed:
Merge branch 'taverns-serialization' into BannerComponentBinaryPackage

garrettluskey committed:
Merge branch 'BannerComponentBinaryPackage' of https://github.com/Bannerlord-Coop-Team/BannerlordCoop into BannerComponentBinaryPackage

garrettluskey committed:
Merge pull request #414 from Bannerlord-Coop-Team/BannerComponentBinaryPackage

BannerComponentBinaryPackage

garrettluskey committed:
Initial commit

garrettluskey committed:
Merge branch 'taverns-serialization' into LordPartyComponentBinaryPackage

garrettluskey committed:
renamed test

garrettluskey committed:
Initial commit

garrettluskey committed:
Update ItemObjectSerializationTest.cs

garrettluskey committed:
Initial commit

garrettluskey committed:
initial commit

garrettluskey committed:
Initial commit

EgardA committed:
CharacterPerks

garrettluskey committed:
Merge pull request #415 from Bannerlord-Coop-Team/LordPartyComponentBinaryPackage

LordPartyComponentBinaryPackage

garrettluskey committed:
Merge pull request #416 from Bannerlord-Coop-Team/CaravanPartyComponentBinaryPackage

 CaravanPartyComponentBinaryPackage

garrettluskey committed:
Merge pull request #417 from Bannerlord-Coop-Team/ItemObjectBinaryPackageUpdate

ItemObjectBinaryPackageUpdate

garrettluskey committed:
Merge branch 'taverns-serialization' into PartyAIBinaryPackage

garrettluskey committed:
Merge pull request #418 from Bannerlord-Coop-Team/PartyAIBinaryPackage

PartyAIBinaryPackage

garrettluskey committed:
Merge branch 'taverns-serialization' into HeroDeveloperBinaryPackage

garrettluskey committed:
Merge pull request #419 from Bannerlord-Coop-Team/HeroDeveloperBinaryPackage

HeroDeveloperBinaryPackage

garrettluskey committed:
Merge branch 'taverns-serialization' into WorkshopBinaryPackage

garrettluskey committed:
Merge pull request #420 from Bannerlord-Coop-Team/WorkshopBinaryPackage

WorkshopBinaryPackage

EgardA committed:
Stuff

EgardA committed:
RegisterUpdate

EgardA committed:
Updated all objects

garrettluskey committed:
Merge branch 'taverns-serialization' into CharactherPerksSerial

garrettluskey committed:
Merge pull request #421 from Bannerlord-Coop-Team/CharactherPerksSerial

Character perks serialization

EgardA committed:
my bad

zzzzzzzzott committed:
Resolves #380

garrettluskey committed:
Update PartyBaseBinaryPackage.cs

garrettluskey committed:
Merge pull request #422 from zzzzzzzzott/taverns-serialization

PartyBase Serialization

garrettluskey committed:
Initial commit

garrettluskey committed:
Added testing bootstrap

garrettluskey committed:
Merge branch 'taverns-serialization' into CharacterAttributeBinaryPackage

garrettluskey committed:
Update CharacterAttributeSerializationTest.cs

garrettluskey committed:
Small improvements/fixes

garrettluskey committed:
Merge branch 'taverns-serialization' into CharacterAttributeBinaryPackage

garrettluskey committed:
Update GameInterface.csproj

garrettluskey committed:
Initial commit

zzzzzzzzott committed:
Resolves #393, #366, #381

Also introduces non-issue binary packages - ExplainedNumber and PathFaceRecord.

garrettluskey committed:
Merge pull request #423 from Bannerlord-Coop-Team/CharacterAttributeBinaryPackage

Created CharacterAttributeBinaryPackage

garrettluskey committed:
Merge branch 'taverns-serialization' into CommonAreaPartyComponent

garrettluskey committed:
Merge pull request #424 from Bannerlord-Coop-Team/CommonAreaPartyComponent

Created CommonAreaPartyComponentBinaryPackage and CommonAreaBinaryPackage

garrettluskey committed:
Cleaned up MobileParty and Kingdom

Added new Bootstrap capability and reorganized it's folder structure

Cleaned up MobileParty serialization and test

Cleaned up Kingdom

garrettluskey committed:
Cleanup

garrettluskey committed:
Merge branch 'taverns-serialization' into pr/425

garrettluskey committed:
Army cleanup

garrettluskey committed:
Merge pull request #425 from zzzzzzzzott/taverns-serialization

Created KingdomBinaryPackage, MobilePartyBinaryPackage, and ArmyBinaryPackage. Also two binary packages that are non-issue, ExplainedNumberBinaryPackage and PathFaceRecordBinaryPackage.

garrettluskey committed:
Initial commit

garrettluskey committed:
Merge branch 'taverns-serialization' into HeroBinaryPackage

garrettluskey committed:
Initial commit

garrettluskey committed:
Merge branch 'CharacterAttributesBinaryPackage' into HeroBinaryPackage

garrettluskey committed:
Initial commit

garrettluskey committed:
Merge pull request #427 from Bannerlord-Coop-Team/CharacterAttributesBinaryPackage

Created CharacterAttributesBinaryPackage

garrettluskey committed:
Added class summary

garrettluskey committed:
Initial commit

garrettluskey committed:
Merge pull request #429 from Bannerlord-Coop-Team/CharacterObjectBinaryPackage

Created CharacterObjectBinaryPackage

zzzzzzzzott committed:
Resolves #370

zzzzzzzzott committed:
Merge branch 'taverns-serialization' of https://github.com/zzzzzzzzott/BannerlordCoop into taverns-serialization

garrettluskey committed:
Merge branch 'taverns-serialization' into HeroBinaryPackage

garrettluskey committed:
Merge pull request #428 from Bannerlord-Coop-Team/HeroBinaryPackage

Created HeroBinaryPackage

garrettluskey committed:
Merge branch 'taverns-serialization' into pr/430

garrettluskey committed:
All Tests passing

garrettluskey committed:
Minor fixes

garrettluskey committed:
Merge pull request #430 from zzzzzzzzott/taverns-serialization

Updated ClanBinaryPackage

garrettluskey committed:
Removed old serializers

garrettluskey committed:
Merge branch 'taverns-serialization' of https://github.com/Bannerlord-Coop-Team/BannerlordCoop into taverns-serialization

garrettluskey committed:
Merge pull request #431 from Bannerlord-Coop-Team/taverns-serialization

Serialization Feature

garrettluskey committed:
Added CharacterObject transfer, release config now working, and misc bug fixes

garrettluskey committed:
Fixed deploy project

garrettluskey committed:
Update MissionTestMod.csproj.user

garrettluskey committed:
Update CharacterObjectBinaryPackage.cs

garrettluskey committed:
Update BinaryPackageBase.cs

garrettluskey committed:
Added agent detection

garrettluskey committed:
Merge pull request #434 from Bannerlord-Coop-Team/taverns-serialization

Taverns CharacterObject Transfer

garrettluskey committed:
Update AgentInteractionPatch.cs

garrettluskey committed:
Merge pull request #435 from Bannerlord-Coop-Team/disable-board-games

Disabled Player Interactions Due To Bug

garrettluskey committed:
Fix for common project issue. It got bricked or something so remade

smudge202 committed:
clear down redundant usings (114 compiler warnings)

smudge202 committed:
remove additional redundant usings

smudge202 committed:
remove dupe reference to MaterialPropertySerializationTest

smudge202 committed:
switch projects with references to taleworlds to x64 builds

smudge202 committed:
remove redundant nlog and stateless references

smudge202 committed:
correct assertion argument order (expect, actual)

garrettluskey committed:
Merge pull request #437 from smudge202/compile-warning-fixes

Compile warning fixes

EgardA committed:
Forfeit fix

EgardA committed:
boardgame updates

EgardA committed:
Improved interactions

EgardA committed:
small update

EgardA committed:
Fixed broken captures

EgardA committed:
Revert "Fix for common project issue. It got bricked or something so remade"

This reverts commit 3c0702cd545f3c5804cc4f849014302958cf9a0c.

EgardA committed:
Cleanup

EgardA committed:
Update BoardGamePatches.cs

garrettluskey committed:
Updated modules to include newly added taleworld modules

garrettluskey committed:
Merge pull request #444 from Bannerlord-Coop-Team/modules-bug-fix

Modules Bug Fix

smudge202 committed:
#440 logging corrections (#441)

* adds configuration and di based server startup

* consolidate nuget package versions

* correct Jokes release config file path

* improve server and client logging

* add batch logging common utility and client network receive logging

* add comments to BatchLogger

garrettluskey committed:
Added network config to NetManager instantiations

garrettluskey committed:
Merge pull request #448 from Bannerlord-Coop-Team/network-config-update

Network Config Update

garrettluskey committed:
Taverns 1.1 kickoff

garrettluskey committed:
Merge branch 'taverns-vertical-slice' of https://github.com/Bannerlord-Coop-Team/BannerlordCoop into taverns-vertical-slice

garrettluskey committed:
Update LiteNetP2PClient.cs

garrettluskey committed:
Naming updates

garrettluskey committed:
Update LiteNetP2PClient.cs

garrettluskey committed:
Merge pull request #450 from Bannerlord-Coop-Team/taverns-1.1-kickoff

Taverns 1.1 kickoff

smudge202 committed:
switch nlog to serilog

smudge202 committed:
prevent cancellation of batch log poller if GC has collected cts

EgardA committed:
Update BoardGamePatches.cs

EgardA committed:
Merge pull request #442 from Bannerlord-Coop-Team/boardgamesfix

BoardGame Fix & Interaction Change

garrettluskey committed:
Merge commit

garrettluskey committed:
Merge branch 'taverns-vertical-slice' into 447-serilog

garrettluskey committed:
Merge pull request #452 from smudge202/447-serilog

switch nlog to serilog

garrettluskey committed:
Merge branch 'taverns-vertical-slice' into taverns-MessageBrokerRefactor

EgardA committed:
pause patch

EgardA committed:
pause patch

garrettluskey committed:
Agents not working

garrettluskey committed:
Serializer bug fix

EgardA committed:
Merge pull request #455 from Bannerlord-Coop-Team/boardgamesfix

pause patch

zzzzzzzzott committed:
Revolves #456

garrettluskey committed:
Bug fix for multi client spawning

smudge202 committed:
correct kingdom deserialization (#459)

* correct kingdom deserialization

* improve agent spawn logging

zzzzzzzzott committed:
Moved CampatibilityInfoBinaryPackge into new internal folder

Removed packages.config

garrettluskey committed:
Fixed formatting errors

garrettluskey committed:
Merge branch 'taverns-MessageBrokerRefactor' of https://github.com/Bannerlord-Coop-Team/BannerlordCoop into taverns-MessageBrokerRefactor

zzzzzzzzott committed:
Patch

zzzzzzzzott committed:
Merge pull request #461 from zzzzzzzzott/taverns-vertical-slice

Move CompatabilityInfo to GameInterface

EgardA committed:
Board game fix

garrettluskey committed:
Merge pull request #465 from Bannerlord-Coop-Team/taverns-MessageBrokerRefactor

Message Broker Refactor

EgardA committed:
Merge branch 'taverns-vertical-slice' into boardgamesfix

EgardA committed:
One event using MessageBroker

EgardA committed:
All events using new message broker

EgardA committed:
Updates based on comments

EgardA committed:
Merge pull request #460 from Bannerlord-Coop-Team/boardgamesfix

Added missing opponent detection & more

zzzzzzzzott committed:
Resolves #462

EgardA committed:
Reord and build fix

EgardA committed:
missing files?

zzzzzzzzott committed:
Merge pull request #472 from zzzzzzzzott/taverns-vertical-slice

Rename Serialization/Impl to Serialization/External

garrettluskey committed:
Moved common to .NET Standard

garrettluskey committed:
Merge pull request #474 from Bannerlord-Coop-Team/common-project-update

Updated Common to NetStandard 2.0

garrettluskey committed:
Minor reorg

garrettluskey committed:
Merge branch 'taverns-vertical-slice' into Reorg

garrettluskey committed:
Updated namespaces

garrettluskey committed:
Updated namespaces

garrettluskey committed:
Updated namespace and imports

garrettluskey committed:
Removed unused import

garrettluskey committed:
better reorg and namespace/using updates

garrettluskey committed:
updated pausing patch location

EgardA committed:
Merge pull request #473 from Bannerlord-Coop-Team/Reorg

Reorganization

garrettluskey committed:
removed old common

garrettluskey committed:
Merge branch 'taverns-vertical-slice' into taverns-vertical-slice-merge

garrettluskey committed:
Updated Common

garrettluskey committed:
Coop project fix

garrettluskey committed:
Merge fixes

garrettluskey committed:
Merge branch 'taverns-vertical-slice-merge' into client-connection-states

garrettluskey committed:
Added minor integration updates

garrettluskey committed:
PacketManager tests and network structure update

garrettluskey committed:
All tests passing once again

garrettluskey committed:
Fixed container failures

garrettluskey committed:
GameInterface refactor

garrettluskey committed:
Merge branch 'development' into client-connection-states

garrettluskey commented:
Class level comments are deferred to #487 and #488

garrettluskey committed:
removed standalone logic interfaces

garrettluskey committed:
Added logging
